packages/realm/src/Object.ts

```diff
@@ -43,6 +43,8 @@ const PROXY_HANDLER: ProxyHandler<RealmObject<any>> = {
   },
 };
 
+type JSONCacheMap = Map<string, Map<string, DefaultObject>>;
```

Comment:
I am defining a type here for readability, not sure if this is meant to be done somewhere / removed.
I am also not sure if maps are best here because they do have to be initialised

Reply:
Moving it to a separate file makes sense in my opinion.